##Q2解説

変更点、追加した箇所の解説です。

追加
```python
# N -> タイムアウトがN回以上続いたときに故障とみなす値
N = 5
```
Nをプログラムのパラメータとして与えられるようにするために追加しました。

変更
```python
# 何回連続してタイムアウトしたかを計算
        # count -> 何回連続してタイムアウトしたかをカウント
        count = 0
        for j in range(len(same_address_df)):
            try:
                # same_address_df[index + j, response_result] -> j個先のResponse_resultの値
                # 基準の位置からj個先がタイムアウトし、基準の位置からすぐ下もタイムアウトしているときカウントに＋１
                if '-' == same_address_df.iat[index + j, response_result] == same_address_df.iat[index + 1, response_result]:
                    count += 1
            except IndexError:
                break

        # count >= N -> N回以上連続してタイムアウトになったときの処理を記述
        if count >= N and '-' != same_address_df.iat[index - 1, response_result]:
            # same_address_df.iat[index + N-1, datetime] -> 最後にタイムアウトした日時
            # same_address_df.iat[index, datetime] -> 最初にタイムアウトした日時
            timeouted_period = same_address_df.iat[index + count - 1, datetime] - same_address_df.iat[
                index, datetime]
            # .txtに出力する処理
            try:
                # format(str, '0>14') -> 日時を０で埋める (YYYYMMDDhhmmssの形式にする)
                f.write(timeouted_server_address + ',')
                f.write(format(timeouted_period, '0>14') + '\n')
            except NameError:
                break
```
変更点は、一つ目のtry句内のelif句を削除したことです。
N回タイムアウトが連続したときを故障とするので、不要になったために削除しました。
<br>
同様に、26行目以降では、タイムアウトの数がN回以上連続したときのみという条件を増やしています。


